puppet: Add a daily cron-job to send zulip update announcements.

prakhar1144 opened the pull request:
Commit 1: Group DM text update.
Commit 2: A daily cron job is configured to run the `send_zulip_update_announcements` management command.

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented:
We might want this cron job to run hourly, maybe at some minute offset late in the hour; I think having notices about a new release come the day after the update may feel bad.

Hmm, I guess the other option would be to add a call to this in the post-upgrade tooling, so that new notices go out immediately, and leave the cron job as daily. Maybe worth discussion in the design thread.

alya commented:
Yeah, I was wondering if there are technical considerations that make daily (rather than hourly) work better. Hourly seems nicer from a UX perspective.

We still want the day of delay between the initial DM and the first notice, though.

prakhar1144 commented:
Update in the recent push:

* Updated text as we discussed in CZO
  <details><summary>zulip_update_announcements_stream configured</summary>
  <img src="https://github.com/zulip/zulip/assets/56781761/21159d9b-74d8-4b40-926a-08f5bf20339e">
  </img>
  </details> 
  
  <details><summary>zulip_update_announcements_stream not configured</summary>
  <img src="https://github.com/zulip/zulip/assets/56781761/859a8862-c24d-465d-9297-9251d5f2383f">
  </img>
  </details> 

* Updated cron job to run hourly.

prakhar1144 commented:
> Hmm, I guess the other option would be to add a call to this in the post-upgrade tooling, so that new notices go out immediately, and leave the cron job as daily. Maybe worth discussion in the design thread.

Updating with this approach shortly.

timabbott commented:
Oops missed that last comment -- I moved the cron job to a different minute (0 is usually a bad choice; you want something random so that one doesn't stack up lots of jobs on the same hour), and merged this, thanks @prakhar1144!

I think we should still look at adding it to the post-upgrade tooling; that can be a follow-up, though; I wanted to get this in so we can test it end-to-end on chat.zulip.org.
